pkg/api/routes.go

```diff
@@ -621,6 +622,26 @@ func (rh *RouteHandler) GetBlob(response http.ResponseWriter, request *http.Requ
 
 	mediaType := request.Header.Get("Accept")
 
+	digestObj, err := godigest.Parse(digest)
+	if err != nil {
+		WriteJSON(response,
+			http.StatusBadRequest,
+			NewErrorList(NewError(DIGEST_INVALID, map[string]string{"digest": digest})))
+	}
+
+	if rh.c.Config.Storage.StorageDriver != nil {
+		redirect, ok := rh.c.Config.Storage.StorageDriver["redirect"].(bool)
+
+		if ok && redirect {
+			path := imgStore.BlobPath(name, digestObj)
+			if path, err := imgStore.URLForPath(path); err == nil {
```

Comment:
please log the error in case there is one :D, it could be. and return a http.StatusInternalServerError

Reply:
I've addressed this comment. I'll pay more attention to these details :D